hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java

```diff
@@ -1606,13 +1606,8 @@ public R getSingleResult() {
 
 	@Override
 	public int executeUpdate() throws HibernateException {
-		if ( ! getProducer().isTransactionInProgress() ) {
-			throw getProducer().getExceptionConverter().convert(
-					new TransactionRequiredException(
-							"Executing an update/delete query"
-					)
-			);
-		}
+		getProducer().checkTransactionNeededForUpdateOperation( "Executing an update/delete query", true );
+
```

Comment:
This above is the only time that ```#checkTransactionNeededForUpdateOperation``` is called with flag for converting the exception set to ```true```.

I see that the original code converts the exception, but I'm not sure if it is necessary to do so, since ```TransactionRequiredException``` is a JPA exception. All that calling ```ExceptionConverter#convert(RuntimeException)``` will do is mark the transaction for rollback. Since we know there is no transaction in progress, then that is a no-op.

I'm curious why the exception needs to be converted here, but not in the other places.

There is no Javadoc for ```ExceptionConverter#convert(RuntimeException)``` to indicate when the method should or should not be called, so that doesn't give guidance.

It seems that when ```#checkTransactionNeededForUpdateOperation``` is called, it should always be called with ```true``` or always with ```false```. To simplify things, I would prefer that ```#checkTransactionNeededForUpdateOperation``` not have the ```convertException``` argument, and that the method would do the right thing consistently.

Reply:
I checked and if I remove the conversion `org.hibernate.test.jpa.compliance.tck2_2.QueryApiTest#testUpdateRequiresTxn` fails.